packages/ActionBar/src/components/ColumnsArrangement/ColumnsArrangement.js

```diff
@@ -120,23 +120,32 @@ export default function ColumnsArrangement(props) {
             I18n.t("actionBar.no_results")
           ) : (
             <sc.Sortable onChange={handleChangeOrder} hasNumbers={false}>
-              {filteredColumnIds.map(id => (
-                <Sortable.Item
-                  key={id}
-                  sortId={id}
-                  isDragDisabled={columns[id].isDisabled}
-                  handleElement={columns[id].isDisabled ? <sc.LockIcon /> : undefined}
-                >
-                  <ColumnManagingItem
+              {filteredColumnIds.map(id => {
+                if (!columns[id]) {
+                  console.error(
+                    `Failed from rendering the ${id} column. Have you passed the <ColumnsArrangement.ColumnDefinition /> for that column?`
+                  );
+                  return null;
+                }
+
+                return (
+                  <Sortable.Item
                     key={id}
-                    id={id}
-                    label={columns[id].label}
-                    isDisabled={columns[id].isDisabled}
-                    isHidden={columns[id].isHidden}
-                    onChangeVisibility={onChangeVisibility}
-                  />
-                </Sortable.Item>
-              ))}
+                    sortId={id}
+                    isDragDisabled={columns[id].isDisabled}
+                    handleElement={columns[id].isDisabled ? <sc.LockIcon /> : undefined}
+                  >
+                    <ColumnManagingItem
+                      key={id}
+                      id={id}
+                      label={columns[id].label}
+                      isDisabled={columns[id].isDisabled}
+                      isHidden={columns[id].isHidden}
+                      onChangeVisibility={onChangeVisibility}
+                    />
+                  </Sortable.Item>
+                );
+              })}
```

Comment:
The only change in this part is adding a `console.error` for the ids available in `orderedColumnIds ` but not available in `<ColumnsArrangement.ColumnDefinition />` . If it happens the other way around, there won't be any error, since we don't use `ColumnDefinition` to iterate and render anything.